src/Message/UpdateTransactionRequest.php

```diff
@@ -33,34 +33,18 @@ public function sendData($data)
 
         // Once the checkout order has reached it's end-state it cannot be changed; update the management order instead
         if (isset($responseData['error_code']) && 'READ_ONLY_ORDER' === $responseData['error_code']) {
-            $dataKeysByManagementEndpoint = [
-                'customer-details' =>  ['shipping_address', 'billing_address'],
-                'merchant-references' => ['merchant_reference1', 'merchant_reference2'],
-            ];
+            // Attempt to update the merchant references at the order management endpoint
+            $requestData = array_intersect_key($data, ['merchant_reference1' => true, 'merchant_reference2' => true]);
 
-            foreach ($dataKeysByManagementEndpoint as $endpoint => $dataKeys) {
-                // Extract the data relevant to the current management endpoint
-                $requestData = array_intersect_key($data, array_flip($dataKeys));
-
-                // Don't perform an unnessasary request if there is no data to be updated for this management endpoint
-                if (empty($requestData)) {
-                    continue;
-                }
-
-                // Send the relevant data to the management endpoint
-                $responseData = $this->getResponseBody(
+            $responseData = !empty($requestData) ?
```

Comment:
Would it not be better to return the original (error) response from the first request in this case, instead of overriding it with an empty array? That was the original behaviour. In this case, you would not have any indication the request has failed

Reply:
That was indeed the old behavior, but that behavior was actually incorrect. When the original request failed because the checkout order became read-only, we should still return a successful response because we fell back to the management API for updating the merchant references.

If we pass along the response data of the checkout API request, then successful fallback to the management API would be interpreted as unsuccessful (`UpdateTransactionResponse::isSuccessful`).